packages/chart/src/CdcChartComponent.tsx

```diff
@@ -159,9 +159,17 @@ const CdcChart: React.FC<CdcChartProps> = ({
     (config.xAxis || config.yAxis) && ['date-time', 'date'].includes((config.xAxis || config.yAxis).type)
   const dataTableDefaultSortBy = hasDateAxis && config.xAxis.dataKey
 
-  const checkLineToBarGraph = () => {
-    return isConvertLineToBarGraph(config.visualizationType, filteredData, config.allowLineToBarGraph)
-  }
+  const convertLineToBarGraph = (() => {
+    const isDynamicSeries = configObj.series.some(series => series.dynamicCategory)
+    const isDynamicWithLessThanThreePoints =
+      isDynamicSeries && _.uniq(configObj.formattedData?.map(data => data[configObj.xAxis.dataKey])).length <= 2
+    return isConvertLineToBarGraph(
+      config.visualizationType,
+      filteredData,
+      config.allowLineToBarGraph,
+      isDynamicWithLessThanThreePoints
+    )
+  })()
```

Comment:
can you put all of this logic into the isConvertLineToBarGraph function? then you can just call the function like const convertLineToBarGraph = isConvertLineToBarGraph(...)